### PR TITLE
Php 7.4

### PR DESCRIPTION
**Pull request by jakubenglicky**
- Upgrade dependencies
- Fix PHPStan
- Fix CodeStyle

### CODE REVIEW

File: src/Auth/AccessTokenCacheProvider.php [hunk 1]
```diff
@@ -26,7 +26,7 @@ public function __construct(IHttpClient $client, IStorage $storage)
 		$this->cache = new Cache($storage, self::CACHE_NAMESPACE);
 	}
 
-	public function getAccessToken(Config $config): AccessToken
```
Comment (f3l1x) [inline src/Auth/AccessTokenCacheProvider.php]: Hold on, it's a BC break. Why is that?

File: src/Client/MessageClient.php [hunk 2]
```diff
@@ -23,7 +23,11 @@ public function send(Message $message): object
 		$this->assertResponse($response, 201);
 
 		$res = Json::decode($response->getBody()->getContents());
-		$res->parsedId = str_replace('messages/', '', Strings::match($res->link, '~messages/\d+~')[0]);
+
+		$res->parsedId = $res->link;
+		if (isset(Strings::match($res->link, '~messages/\d+~')[0])) {
+			$res->parseId = str_replace('messages/', '', Strings::match($res->link, '~messages/\d+~')[0]);
```
Comment (f3l1x) [inline src/Client/MessageClient.php]: What about  `Strings::match($res->link, '~messages/\d+~')[0] ?? ''`.

File: src/Auth/AccessTokenCacheProvider.php (hunk 1 above)
Comment (jakubenglicky) [reply src/Auth/AccessTokenCacheProvider.php]: In this method is possible that return statement is NULL, but the return type wasn't nullable and PHPStan was sad.

Comment (f3l1x) [reply src/Auth/AccessTokenCacheProvider.php]: Hmm, we can't change a notation. Better to throw an exception or sth like that.

**Comment by f3l1x** [comment]
ping @jakubenglicky

**Comment by f3l1x** [comment]
ping @jakubenglicky ;-)

**Comment by jakubenglicky** [comment]
> ping @jakubenglicky ;-)

Yeah, I'm just getting ready for it

**Comment by jakubenglicky** [comment]
Felix, PR is ready for review.

File: src/Client/MessageClient.php [hunk 3]
```diff
@@ -23,7 +23,8 @@ public function send(Message $message): object
 		$this->assertResponse($response, 201);
 
 		$res = Json::decode($response->getBody()->getContents());
-		$res->parsedId = str_replace('messages/', '', Strings::match($res->link, '~messages/\d+~')[0]);
+
+		$res->parseId = str_replace('messages/', '', Strings::match($res->link, '~messages/\d+~')[0] ?? '');
```
Comment (vody105) [inline src/Client/MessageClient.php]: neni toto typo? parsedId -> parseId?

**Review by vody105 — CHANGES_REQUESTED** [review]

File: src/Client/MessageClient.php (hunk 3 above)
Comment (jakubenglicky) [reply src/Client/MessageClient.php]: To je, máš dobrý :eye: :wink:

**Review by vody105 — APPROVED** [review]

**Comment by f3l1x** [comment]
Awesome, thanks @jakubenglicky

**Pull request merged**